docs/plugin-api.rst

```diff
@@ -130,3 +130,41 @@ language
   The language determined from the class attribute (or undefined).
 
 It returns nothing.
+
+
+before:compile(mode, parent)
```

Comment:
There are actially `before:compileEarly` and `before:compileLate`, right?

Reply:
I need to rework all the docs (which is why this isn't merged).  No one has asked for this as a plugin so I'm probably going to rip that out completely (plus I'm not sure a world where to install grammar xyz you also need plugin 123 and 234 is a great place to end up, so we can push off that decision into the future)... so the new capability is just that languages can define their own sugar... and then inside the library we can "test" new extensions in a single language first... and if that goes well they can be promoted to the compiler.

So these docs are probably moving into `compilerExtensions` inside the mode docs.

Reply:
Moved docs into the mode reference.